File: src/main/java/com/sber/yamlprocessor/model/Stage.java
package com.sber.yamlprocessor.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;

public class Stage {
    
    private Boolean id;
    
    @NotNull(message = "Executor is required")
    private String executor;
    
    @Valid
    private Log log;
    
    @JsonProperty("context-code")
    private ContextCodes contextCode;
    
    @NotNull(message = "Description is required")
    private String description;
    
    @Valid
    @NotNull(message = "Configurator is required")
    private Configurator configurator;

    public Stage() {}

    public Stage(String executor, String description, Configurator configurator) {
        this.executor = executor;
        this.description = description;
        this.configurator = configurator;
    }

    // Getters and Setters
    public Boolean getId() {
        return id;
    }

    public void setId(Boolean id) {
        this.id = id;
    }

    public String getExecutor() {
        return executor;
    }

    public void setExecutor(String executor) {
        this.executor = executor;
    }

    public Log getLog() {
        return log;
    }

    public void setLog(Log log) {
        this.log = log;
    }

    public ContextCodes getContextCode() {
        return contextCode;
    }

    public void setContextCode(ContextCodes contextCode) {
        this.contextCode = contextCode;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Configurator getConfigurator() {
        return configurator;
    }

    public void setConfigurator(Configurator configurator) {
        this.configurator = configurator;
    }
}

File: src/main/kotlin/com/sber/yamlprocessor/model/Stage.kt
package com.sber.yamlprocessor.model

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.persistence.CascadeType
import jakarta.persistence.Column
import jakarta.persistence.Embedded
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.OneToOne
import jakarta.persistence.Table

@Entity
@Table(name = "stage")
class Stage(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonIgnore
    @Column(name = "id")
    var dbId: Long? = null,

    @ManyToOne(optional = false)
    @JoinColumn(name = "subprocess_id")
    @JsonIgnore
    var subprocess: Subprocess? = null,

    @Column(name = "stage_bool_id")
    var id: Boolean? = null,

    @Column(name = "executor", nullable = false, length = 500)
    var executor: String = "",

    @Embedded
    var log: Log? = null,

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "context_code", referencedColumnName = "code")
    @JsonProperty("context-code")
    var contextCode: ContextCodesDictionary? = null,

    @Column(name = "description", nullable = false, length = 4000)
    var description: String = "",

    @OneToOne(cascade = [CascadeType.ALL], orphanRemoval = true)
    @JoinColumn(name = "configurator_id")
    var configurator: Configurator? = null
)
